src/algo_tx.c

```diff
@@ -160,6 +160,38 @@ map_kv_bin(uint8_t **p, uint8_t *e, char *key, uint8_t *valbuf, int vallen)
   return 1;
 }
 
+static int
+map_kv_params(uint8_t **p, uint8_t *e, char *key, struct asset_params *params)
+{
+  // Save original buffer in case we end up with a zero value
+  uint8_t *psave = *p;
+
+  encode_str(p, e, key, SIZE_MAX);
+
+  uint8_t *mapbase = *p;
+  *((*p)++) = FIXMAP_0;
```

Comment:
Do you need to check that `*p < e` here?

Reply:
Maybe use `put_byte`?

Reply:
Good catch, thanks!